### PR TITLE
[INSIGHTS-13] Add Algorithmia Insights ability

### PR DESCRIPTION
**Pull request by dinoboy197**
* Adds the Insights API to the R library per [the updates to the R client documentation for Insights](https://github.com/algorithmiaio/dev-center/pull/349).
* Makes tests run correctly (they were not running correctly before, so failures were silent!)
* Adds CI for running tests and dry-run of submitting to CRAN using RHub
* Adds documentation for releases to CRAN
* Comments out a couple of broken tests ([JIRA filed](https://algorithmia.atlassian.net/browse/DEV-74) to fix and re-enable)

This has been tested by:
* adding a true CI build of the R client, with:
  * running tests in CI
  * using R's recommended approach to verifying a packaged submission against CRAN
* insights1 testing cluster and correctly pushes insights to the platform (and out to Kafka) with algorithm https://insights1.enthalpy.click/algorithms/algorithmia_admin/traacksept28r

### CODE REVIEW

**Comment by dinoboy197** [comment]
Moving back to draft as I am add some CI checks and getting things ready for CRAN submission.

**Comment by dinoboy197** [comment]
One pre-existing test (`loaderThrowsException`) is failing now that CI is enabled for tests. [Pinged](https://algorithmia.slack.com/archives/C01BBU6TRT2/p1601664825019900) @zeryx for assistance.

File: tools/build-cran-package [hunk 1]
```diff
@@ -1,34 +0,0 @@
-#!/bin/bash
```
Comment (dinoboy197) [inline tools/build-cran-package:1]: This file has been changed to `test-cran-package`, and now uses RHub to run CRAN checks.

File: tests/DataFileTest.r [hunk 2]
```diff
@@ -46,6 +46,10 @@ test.deleteNonExistantFile <-function() {
 
 test.fileCreationAndDeletion <- function() {
   client <- getAlgorithmiaClient(Sys.getenv("ALGORITHMIA_API_KEY", unset=NA))
+  dataDir <- client$dir("data://.my/largeFiles")
```
Comment (dinoboy197) [inline tests/DataFileTest.r:49]: Making these tests more robust by ensuring that they have the directory they need beforehand.

File: tests/RunAllTests.r [hunk 3]
```diff
@@ -1,16 +1,36 @@
-# Uncomment these lines to test local (unpublished) changes to the client
```
Comment (dinoboy197) [inline tests/RunAllTests.r:1]: This test file has been modified to actually run the tests correctly (it didn't actually run the test code before :fearful: ) and then fail if any errors were reported. RUnit is pretty old and no longer maintained; they should probably be ported to a more modern testing framework in the future.

File: R/Client.r [hunk 4]
```diff
@@ -40,6 +40,14 @@ AlgorithmiaClient <- methods::setRefClass("AlgorithmiaClient",
        AlgorithmiaDataDirectory objects are used to interact with directories."
       getDataDirectory(.self, dataUrl)
     },
+    reportInsights = function(insights) {
```
Comment (dinoboy197) [inline R/Client.r:43]: This is the one new method being added in this PR.

File: DESCRIPTION [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
 Type: Package
 Title: Allows you to Easily Interact with the Algorithmia Platform
-Version: 0.2.0
-Date: 2019-07-31
-Author: James Sutton, James Athappilly
-Maintainer: James Athappilly <jamesa@algorithmia.com>
+Version: 0.3.0
+Date: 2020-09-28
+Author: James Sutton, James Athappilly, Taylor Raack
+Maintainer: Robert Fulton <rfulton@algorithmia.com>
```
Comment (dinoboy197) [inline DESCRIPTION:7]: Robert Fulton is a special user account set up to forward email; talk to Taylor about this with questions.

Comment (zeryx) [reply DESCRIPTION:7]: It's hilarious that we have to do this.. CRAN is weird haha

File: tests/AlgorithmHandlerTest.r [hunk 6]
```diff
@@ -124,7 +124,7 @@ test.arityProblemWithContextThrowsException <- function() {
   checkEquals(result, expected)
 }
 
-test.loaderThrowsException <- function() {
+disabledtest.loaderThrowsException <- function() {
```
Comment (dinoboy197) [inline tests/AlgorithmHandlerTest.r:127]: Test does not pass, and cannot figure out what is wrong here - created [DEV-74](https://algorithmia.atlassian.net/browse/DEV-74) to review and fix.

File: tests/AlgorithmTest.r [hunk 7]
```diff
@@ -67,7 +67,7 @@ test.runWithIntegerInputOutput <-function() {
   checkEquals(algorithm$pipe(10)$result, 10)
 }
 
-test.runWithRawInputOutput <-function() {
+disabledtest.runWithRawInputOutput <-function() {
```
Comment (dinoboy197) [inline tests/AlgorithmTest.r:70]: Tests do not pass, and cannot figure out what is wrong here - created [DEV-74](https://algorithmia.atlassian.net/browse/DEV-74) to review and fix.

**Comment by dinoboy197** [comment]
@algorithmiaio/developer-technologies-contractors ping for review - this is finally ready for you to review and approve! :smile:

**Comment by zeryx** [comment]
Is there no way to test/verify functionality locally @dinoboy197 ? I feel uneasy about oking a build where we can't verify locally, and we had to comment out a number of tests to get things to compile.

**Comment by dinoboy197** [comment]
> Is there no way to test/verify functionality locally @dinoboy197 ?

I think I'd have to ask you for one test:
* `AlgorithmHandlerTest.loaderThrowsException` was authored by you [in this PR](https://github.com/algorithmiaio/algorithmia-r/pull/1)

The other two tests were written by James A who is no longer with us, and never went through PR review, so I'm not sure how he tested those:
* `AlgorithmTest.runWithRawInputOutput` and `AlgorithmTest.runRaw` were written [here](https://github.com/algorithmiaio/algorithmia-r/commit/26e03be5)

It seems they were coded to talk directly to the public marketplace, which isn't a good practice, but unfortunately, that's in the past now.

> I feel uneasy about oking a build where we can't verify locally

Do you mean to say that you don't think that CI running the tests in a clean environment is sufficient to prove that the existing tests work?

> we had to comment out a number of tests to get things to compile

That's not quite accurate. We commented three tests because they did not pass, not because the runtime or test time code didn't compile.

**Review by zeryx — APPROVED** [review]
After speaking with @dinoboy197 we were able to nail down the specific changes, and discovered that our old tests weren't really successful. I agree with the removal of our failure test for now.

**Review by John-Bragg — APPROVED** [review]

**Pull request merged**